### ISSUE TITLE
User exist: add fail, instead of update

### ISSUE THREAD

**Issue by kichetof** [Adldap2/Adldap2#46]
Hi Steve,

Thanks for all your great improvement! :)

I encounter a bug when I launch my form when the user already exist. Normaly, if I think well, `function save` should be update the user if exists.
But for me it doesn't work:

```
<b>Warning</b>:  ldap_add(): Add: Already exists in <b>/vendor/adldap2/adldap2/src/Connections/Ldap.php</b> on line <b>487</b>
``` 

I think that the `$this->exists` on `function setRawAttributes` need to lookup inside AD instead of set by default true ?

```
public function setRawAttributes(array $attributes = [])
    {
        $this->attributes = $attributes;
//        $this->exists = true;
        $this->exists = $this->search()
->orWhere(ActiveDirectory::ACCOUNT_NAME, "=", $this->attributes->getAccountName())
->orWhere(ActiveDirectory::COMMON_NAME, "=", $this->attributes->getCommonName())
->orWhere(ActiveDirectory::FIRST_NAME, "=", $this->attributes->getFirstName())
->orWhere(ActiveDirectory::LAST_NAME, "=", $this->attributes->getLastName())
->orWhere(ActiveDirectory::EMAIL, "=", $this->attributes->getEmail())
        ->get(); // something like that; not tested!!

        $this->syncOriginal();

        return $this;
    }
```

Maybe I'm on the wrong way ? What do you think ?
Many thanks for your great help!

**Comment by stevebauman** [comment]
Yup you're right. Setting the exists attribute blindly to true is probably the wrong way to go here. I'll search the AD by the using the DN attribute and if it can find it then i'll set exists to true. Will be fixed shortly! Thanks!

**Comment by kichetof** [comment]
Thanks a lot ! :)

**Comment by stevebauman** [comment]
Fixed in master, 5.1 will be coming shortly with this fix.

**Comment by kichetof** [comment]
Really great stuff!!! Thanks!

Envoyé de mon iPhone
